rskj-core/src/main/java/co/rsk/peg/P2shErpRedeemScriptBuilder.java

```diff
@@ -0,0 +1,82 @@
+package co.rsk.peg;
+
+import co.rsk.bitcoinj.core.BtcECKey;
+import co.rsk.bitcoinj.core.Utils;
+import co.rsk.bitcoinj.core.VerificationException;
+import co.rsk.bitcoinj.script.Script;
+import co.rsk.bitcoinj.script.ScriptBuilder;
+import co.rsk.bitcoinj.script.ScriptOpCodes;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+
+public class P2shErpRedeemScriptBuilder implements ErpRedeemScriptBuilder{
+    private static final Logger logger = LoggerFactory.getLogger(P2shErpRedeemScriptBuilder.class);
+
+    public Script createRedeemScript(List<BtcECKey> defaultPublicKeys,
+                                     List<BtcECKey> emergencyPublicKeys,
+                                     long csvValue) {
+        Script defaultRedeemScript = ScriptBuilder.createRedeemScript(
+            defaultPublicKeys.size() / 2 + 1,
+            defaultPublicKeys);
+        Script emergencyRedeemScript = ScriptBuilder.createRedeemScript(
+            emergencyPublicKeys.size() / 2 + 1,
+            emergencyPublicKeys);
+
+        byte[] serializedCsvValue = Utils.signedLongToByteArrayLE(csvValue);
+        Script redeemScript = createRedeemScript(defaultRedeemScript, emergencyRedeemScript, serializedCsvValue);
+        validateRedeemScriptValues(defaultRedeemScript, emergencyRedeemScript, csvValue, redeemScript);
+
+        return redeemScript;
+    }
+    public static Script createRedeemScript(Script defaultRedeemScript,
+                                     Script emergencyRedeemScript,
+                                     byte[] serializedCsvValue) {
+
+        ScriptBuilder scriptBuilder = new ScriptBuilder();
+
+        return scriptBuilder
+            .op(ScriptOpCodes.OP_NOTIF)
+            .addChunks(defaultRedeemScript.getChunks())
+            .op(ScriptOpCodes.OP_ELSE)
+            .data(serializedCsvValue)
+            .op(ScriptOpCodes.OP_CHECKSEQUENCEVERIFY)
+            .op(ScriptOpCodes.OP_DROP)
+            .addChunks(emergencyRedeemScript.getChunks())
+            .op(ScriptOpCodes.OP_ENDIF)
+            .build();
+    }
+
+
+    private static void validateRedeemScriptValues(
```

Comment:
This seems to repeat in all builders. Should it be moved to a common place?

Reply:
Good idea